src/phoenix/utilities/ratelimits.py

```diff
@@ -0,0 +1,210 @@
+import asyncio
```

Comment:
I don't think this ratelimits is truely generic as it pertains pretty explicitly to models. I would co-locate this with the models that will be decorated.

Reply:
I think moving this whole module closer to our model interfaces is a good idea.